MiniPosSystem/frmPayment.cs

```diff
@@ -46,11 +47,49 @@ private void BtnSubmit_Click(object sender, EventArgs e)
                 AddNewCard();
             }
 
-            TransactionsDB.AddTransaction(order);
-            MessageBox.Show("Thank you for dining with us!");
-            ActiveForm.Close();
+            if(IsValidInput())
+            {
+                TransactionsDB.AddTransaction(order);
+                MessageBox.Show("Thank you for dining with us!");
+                ActiveForm.Close();
+            }
+
+            btnSubmit.Enabled = true;
+
+        }
+
+
+        /// <summary>
```

Comment:
This IsValidInput() method works for this specific form, but is clearly not great for refactoring support. I'll alter accordingly.